### PR TITLE
Update getMentoringPrograms to filter data by enrolment state

### LINKED ISSUE #129: Update getMentoringPrograms endpoint to return filtered programs on a given mentor state
**Is your feature request related to a problem? Please describe.**
Currently `api/me/programs/mentor` endpoint returns the list of programs the current user has a relationship with and there's no way to filter out the data on the mentor state.

**Describe the solution you'd like**
Update the endpoint to receive a non-required (`required = false`) url parameter to get the mentor state and return data accordingly

### PR DESCRIPTION
**Pull request by Gravewalker666**
## Purpose

The purpose of this PR is to fix #129 

## Goals

- To add the feature of filtering out `my mentoring programs` by the enrolment state

## Approach
- Updated the controller to receive a param of `EnrolmentState`
- Updated the `getMentoringPrograms` function to take the parameter passed down from the controller
- Added a new query function to the `mentorRepository` to get programs by `profileId` and `state`

##  Checklist
- [x] This PR doesn't commit any keys, passwords, tokens, usernames, or other secrets.
- [x] I have read and understood the development best practices guidelines ( http://bit.ly/sef-best-practices )
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented on my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation

## Test environment
- Java Oracle 8
- Apache mvn 3.6.3
- macOS BigSur

### CODE REVIEW

File: src/main/java/org/sefglobal/scholarx/controller/AuthUserController.java [hunk 1]
```diff
@@ -47,10 +47,10 @@ public Object getLoggedUser(Authentication authentication)
 
   @GetMapping("/programs/mentor")
   @ResponseStatus(HttpStatus.OK)
-  public List<Program> getMentoringPrograms(Authentication authentication)
+  public List<Program> getMentoringPrograms(Authentication authentication, @RequestParam(required = false) EnrolmentState state)
```
Comment (YohanAvishke) [inline src/main/java/org/sefglobal/scholarx/controller/AuthUserController.java:50]: break the line into two lines

Comment (YohanAvishke) [inline src/main/java/org/sefglobal/scholarx/controller/AuthUserController.java:50]:
```suggestion
  public List<Program> getMentoringPrograms(Authentication authentication, @RequestParam(required = false) EnrolmentState mentorState)
```
Make the var meaningful

File: src/main/java/org/sefglobal/scholarx/service/IntrospectionService.java [hunk 2]
```diff
@@ -99,19 +99,25 @@ public Profile getLoggedInUser(long id)
      * Retrieves all the {@link Program} objects where the user is a {@link Mentor}
```
Comment (YohanAvishke) [inline src/main/java/org/sefglobal/scholarx/service/IntrospectionService.java:99]: Description is inaccurate due to the new filter

File: src/main/java/org/sefglobal/scholarx/service/IntrospectionService.java [hunk 3]
```diff
@@ -99,19 +99,25 @@ public Profile getLoggedInUser(long id)
...
      *
      * @param id which is the Profile id of the user
+     * @param state which is the enrollment state of the mentor in which the programs are required
```
Comment (YohanAvishke) [inline src/main/java/org/sefglobal/scholarx/service/IntrospectionService.java:102]:
```suggestion
     * @param state of the mentor in which the programs are required
```

**Review by YohanAvishke — CHANGES_REQUESTED** [review]
Great job. Few minor suggestions

File: src/main/java/org/sefglobal/scholarx/controller/AuthUserController.java (hunk 1 above)
Comment (Gravewalker666) [reply src/main/java/org/sefglobal/scholarx/controller/AuthUserController.java:50]: On it

Comment (Gravewalker666) [reply src/main/java/org/sefglobal/scholarx/controller/AuthUserController.java:50]: Sure

File: src/main/java/org/sefglobal/scholarx/service/IntrospectionService.java (hunk 2 above)
Comment (Gravewalker666) [reply src/main/java/org/sefglobal/scholarx/service/IntrospectionService.java:99]: I'll update the description, thanks

File: src/main/java/org/sefglobal/scholarx/service/IntrospectionService.java (hunk 3 above)
Comment (Gravewalker666) [reply src/main/java/org/sefglobal/scholarx/service/IntrospectionService.java:102]: Will do

**Comment by Gravewalker666** [comment]
@YohanAvishke I updated the PR with the requested changes with the commit 439433a
Please review

**Pull request merged**